src/main.js

```diff
@@ -28,9 +28,13 @@ ctx.imageSmoothingEnabled = format.smoothing;
 var metadataList = [];
 var attributesList = [];
 var dnaList = new Set();
-const DNA_DELIMITER = "-";
+const selectedTraitsList = new Set();
 const HashlipsGiffer = require(`${basePath}/modules/HashlipsGiffer.js`);
 
+const { selectTraits, createDna, isDnaUnique, constructLayerToDna, filterDNAOptions } = require('./dna');
+const { needsExclusion } = require('./exclusions');
+const { saveImage } = require('./io');
```

Comment:
some of the code present in this main file was exported to specific modules and the new exclusions functionality is modularised too